cpp/cluster/Cluster.cc

```diff
@@ -25,26 +25,27 @@ Cluster::Cluster(float rcut) : m_rcut(rcut), m_num_particles(0), m_num_clusters(
         throw invalid_argument("Cluster requires that rcut must be non-negative.");
 }
 
-void Cluster::computeClusters(const freud::locality::NeighborQuery* nq,
-                              const freud::locality::NeighborList* nlist, const vec3<float>* points,
-                              unsigned int Np, freud::locality::QueryArgs qargs)
+void Cluster::compute(const freud::locality::NeighborQuery* nq,
+                      const freud::locality::NeighborList* nlist,
+                      const vec3<float> *points,
+                      unsigned int Np,
+                      freud::locality::QueryArgs qargs)
 {
     assert(points);
     assert(Np > 0);
 
     // reallocate the cluster_idx array if the size doesn't match the last one
     if (Np != m_num_particles)
```

Comment:
Does it make sense to remove these guards, since resize is making the same check?

Reply:
Yes, probably. I was actually thinking that I could avoid exposing reallocation to the Cython side at all by just adding a flag that also indicates whether or not reallocation needs to happen, and then every compute class would always just call some function (called `prepArray` or something... names are hard) that would resize if needed, otherwise reallocate if needed. Does that seem reasonable? I don't know if there's any way to automate that process, but at least simplifying the logic to a single function call per compute class would be nice.